containers/src/Kokkos_DualView.hpp

```diff
@@ -804,25 +813,51 @@ class DualView : public ViewTraits<DataType, Properties...> {
     }
   }
 
-  inline void modify_device() {
-    if (impl_dualview_stores_single_view) return;
-
-    if (modified_flags.data() != nullptr) {
-      modified_flags(1) =
-          (modified_flags(1) > modified_flags(0) ? modified_flags(1)
-                                                 : modified_flags(0)) +
-          1;
-      impl_report_device_modification();
+  inline void modify_host() {
+    if constexpr (impl_dualview_stores_single_view) {
+      return;
+    } else {
+      if (modified_flags.data() != nullptr) {
+        modified_flags(0) =
+            (modified_flags(1) > modified_flags(0) ? modified_flags(1)
+                                                   : modified_flags(0)) +
+            1;
+        impl_report_host_modification();
 #ifdef KOKKOS_ENABLE_DEBUG_DUALVIEW_MODIFY_CHECK
-      if (modified_flags(0) && modified_flags(1)) {
-        std::string msg = "Kokkos::DualView::modify_device ERROR: ";
-        msg += "Concurrent modification of host and device views ";
-        msg += "in DualView \"";
-        msg += d_view.label();
-        msg += "\"\n";
-        Kokkos::abort(msg.c_str());
+        if (modified_flags(0) && modified_flags(1)) {
+          std::string msg = "Kokkos::DualView::modify_host ERROR: ";
+          msg += "Concurrent modification of host and device views ";
+          msg += "in DualView \"";
+          msg += d_view.label();
+          msg += "\"\n";
+          Kokkos::abort(msg.c_str());
+        }
+#endif
       }
+    }
+  }
+
+  inline void modify_device() {
+    if constexpr (impl_dualview_stores_single_view) {
+      return;
```

Comment:
hmm ... maybe we should still mark it as modified on the device ... Thinking about consistency and potential migration of memory I would prefer if we keep the marking consistent independent of if it is multiple views or not

Reply:
Can you elaborate? What exactly are your concerns about "migration of memory"? What user-facing changes are you concerned about?

Reply:
Ah, sorry, it was there before this PR.
I was referring to
```
inline void modify_device() {
  if constexpr (impl_dualview_stores_single_view) {
      return;
  }
```
The thought was: do we want to keep on carrying the flags to track modification even for a single underlying view? Even if the `sync` functions basically are no-ops. But thinking about it a bit more, I see, that it makes sense to just disable this tracking as it becomes meaningless for automatically migrating pages